internal/container/set.go

```diff
@@ -0,0 +1,23 @@
+package set
+
+// Set is a map as a set data structure.
+type Set[T comparable] map[T]struct{}
```

Comment:
```suggestion
type Set[T any] map[T]struct{}
```
interface{} can be the element of a map, so is it better?

Reply:
As our notation-go library is still using go 1.18, your suggested change would cause error: "incomparable map key type T (missing comparable constraint)". For more details, please see https://github.com/golang/go/issues/51338.